.editorconfig

```diff
@@ -0,0 +1,18 @@
+# EditorConfig is awesome: http://EditorConfig.org
+
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+insert_final_newline = true
+trim_trailing_whitespace = true
+indent_style = space
+indent_size = 4
+
+[{package.json,.travis.yml}]
+indent_style = space
+indent_size = 2
+
+[Makefile]
+indent_style = tab
```

Comment:
This should indeed be in a separate commit. It deals with many things, not just JS. Should be labelled "provide interoperable editor configuration", for instance